tsc/meetings/2020-04-09.md

```diff
@@ -0,0 +1,106 @@
+2) Secretary was Nick Avramoussis
+
+3) OpenVDB AX recap
+
+Discussion ongoing between Nick and Jeff as various AX features are nailed down.
+A number of proposed changes have been approved -  bitwise operations on
+integer vectors, restricting comparison operation on container types and correct
+implementation of fmod. Two outstanding requests are the beahviour of
+matrix and scalar operations and container (vector) extension. Jeff's preferred
+behaviour for the prior is to treat scalar promotion as an identity matrix
+with its identity components set to the corresponding scalar. For the latter,
+VEX currently extends vectors with a value of one to better match the behaviour
+of various applications against matrices of otherwise incompatible sizes and to
+better correspond to a default value of colour (alpha) or homogeneous coordiante
+(w). The alternative would be to explicitly error on this behaviour. AX
+currently already errors on assignment of containers of mismatching sizes.
+
+4) Nick proposed that every TSC meeting additionally begin with a quick recap
+of newly opened forum posts or github issues. Unanimous consensus. Two topics
+for discussion.
+
+4a) OpenVDB Documentation
+
+New forum post requesting update on OpenVDB documentation. Agreement that
+OpenVDB technical documentation needs a revamp and could benefit from extension.
+This is a significant task with no obvious starting point. Andre mentioned that
+the unit tests have served as a good place to observe various intended usage of
+VDB methods. Using the unit tests would provide a base set of examples to start
+writing documentation. Nick, concern that as the unit tests have not necessarily
+been designed for this that some of the logic in there could be "inaccurate" in
+terms of an optimal implementation solution. Additional care to be taken that
+the unit tests are first and foremost that, with clear documentation of the
+intended example. Nick to create a ticket with a task to identity some of the
+larger tools missing documentation and corresponding unit tests which could be
+used as a first doc attempt.
+
+4b) Hessian computation of VDB grids
+
+Github issue asking if OpenVDB supports the computation of the Hessian matrix
+from a scalar Grid. Currently unsupported. Ken, should be fairly trivial to
+implement. However, as the result per value is a matrix, there is no native way
+to store this information as VDB does not register matrix grids by default.
+Registering these grid types would be one option - however compile and runtime
+variants of "strided" grids (where each voxel/tile value contains an array of
+scalar values of a fixed size) were discussed as alternative solutions which
+could add more flexibility. Some potential confusion when inferring values on
+a grid with an "intrinsic" type i.e. Vec3f vs float[3] collocated vs staggered
+interpretation. General agreement that runtime variants would be interesting to
+pursue. Nick to create a ticket and reply to issue.
```

Comment:
Intriguing, I wonder what the performance implications would be to switch to a runtime equivalent to Vec3f, it sounds like quite a lot of engineering to answer that question. I suppose that could follow the proposal to move implementation out of the hierarchy classes (tree, grid, etc) into dedicated tools to simplify this task, but it's probably still non-trivial. It sounds like the idea of introducing compile-time Mat3/Mat4 variants was discarded?

I'll add the runtime strided grids to the roadmap as well, as it's worth discussing in that context.

Reply:
Not discarded, it was raised.   But then naturally you ask about [6] and other odd-ball sized grids.  Adding four new template expansions (float/double and mat3/mat4) to destroy compile time vs solving it with a runtime tuple solution?

Reply:
Personally, I would prioritize work aimed at reducing compile times generally such as moving implementation out of grid/tree classes as mentioned and introducing explicit template instantiation or some other compile-time registration mechanism, then consider this runtime grid proposal in that new light. If it need not destroy compile times, perhaps that changes how we would make the decision to proceed here.